app/assets/javascripts/discourse/app/lib/swipe-events.js

```diff
@@ -0,0 +1,233 @@
+import { isTesting } from "discourse-common/config/environment";
+
+/**
+   Swipe events is a class that allows components to detect and respond to swipe gestures
+   It sets up custom events for swipestart, swipeend, and swipe for beginning swipe, end swipe, and during swipe. Event returns detail.state with swipe state, and the original event.
+
+   Calling preventDefault() on the swipestart event stops swipe and swipeend events for the current gesture. it is re-enabled on future swipe events.
+**/
+export const SWIPE_DISTANCE_THRESHOLD = 50;
+export const SWIPE_VELOCITY_THRESHOLD = 0.12;
+export const MINIMUM_SWIPE_DISTANCE = 5;
+export const MAX_ANIMATION_TIME = 200;
+export default class SwipeEvents {
```

Comment:
Having this abstracted out into a native class is a massive improvement - nice!

The modern Ember pattern for this kind of thing would be a 'modifier'. Modifiers are attached to DOM elements via hbs templates, and can be passed arguments/callbacks. So for example, this one could look something like


```
<div {{swipe-events onSwipeStart=this.onSwipeStart onSwipe=this.onSwipe onSwipeEnd=this.onSwipeEnd}}>
```
We have a few custom modifiers in core already. e.g. [draggable](https://github.com/discourse/discourse/blob/7bd28db71670047b0b26a5b8c8c54d83945874e2/app/assets/javascripts/discourse/app/modifiers/draggable.js).

However, using that pattern here right now will be quite tricky because `site-header` is extending the MountWidget component and does not have its own hbs template.

So I think what we've got here is a great step in the right direction, and will help us refactor it into a modifier eventually 👌